devel/varlink-libnmstate.md

```diff
@@ -0,0 +1,185 @@
+# Varlink support for libnmstate
+
+## Introduction
+This documentation details the varlink protocol support for libnmstate functions. Varlink protocol encodes all the messages as JSON objects and communicates via unix and tcp socket connections. Using the varlink, libnmstate is accessible via varlink stdin/out and varlink client implementation. This documentation uses python
+programming language for examples, please refer [language binding document](ttps://varlink.org/Language-Bindings#how-to-test-new-language-bindings) for other existing varlink programming languages binding. Libnmstate service is defined as `io.nmstate` in varlink interface address and functions should be called as specified in varlink interface defination `io.nmstate`.
+
+* [Varlink documentation]()
+* [libnmstate API documentation](https://www.nmstate.io/devel/api.html)
+* [Github Repository](https://github.com/nmstate/nmstate/tree/master/libnmstate/ifaces)
+
+## Installation of varlink
+Pypi:
+```
+$ pip3 install -u varlink
+```
+Dnf:
+```
+$ sudo dnf install python3-varlink
+```
+
+## Varlink for service information
+
+To view information about io.nmstate service.
+
+```
+$ varlink info unix:@nmstate
```

Comment:
The goal should be that `varlink info io.nmstate` works once the PR for nmstate is finished.

Reply:
> The goal should be that `varlink info io.nmstate` works once the PR for nmstate is finished.
@tyll 
I order to `varlink info io.nmstate`work it should resolved using varlink resolver ?

Reply:
yes

Reply:
I'll make changes

Reply:
Please focus first on integrating it properly with setup.py, this will then also make it part of the RPM. When this is working, adding the systemd service would be next in my opinion.

Reply:
Okay.